### PR TITLE
drivers: sensor: npm1300: Added charging support without NTC

### PR DESCRIPTION
**Pull request by jasr93**
Added  charging support without NTC for npm1300 charger driver

### CODE REVIEW

**Comment by MaureenHelm** [comment]
@jasr93 please fix the compliance check.

@anangl please take a look

File: drivers/sensor/nordic/npm1300_charger/npm1300_charger.c [hunk 1]
```diff
@@ -456,6 +456,7 @@ int npm1300_charger_init(const struct device *dev)
 {
 	const struct npm1300_charger_config *const config = dev->config;
 	uint16_t idx;
+	uint8_t byte = 0U;
```
Comment (nordic-auko) [inline drivers/sensor/nordic/npm1300_charger/npm1300_charger.c:459]: Nitpick: more descriptive name for this variable.

**Review by nordic-auko — APPROVED** [review]
Looks good. Testing and it is working.

Not required, but nice to have: return -ENOTSUP for `SENSOR_CHAN_GAUGE_TEMP` in `npm1300_charger_channel_get()` when NTC is disabled.

**Review by jaz1-nordic — COMMENTED** [review]
Also merge commits into one.

**Comment by rajeev1986** [comment]
Any plans to get this merge into main? I was thinking this issue is already fixed by Nordic in NCS 2.7.0 but found out that it's not and is still pending. Only realized this after seeing the charger not working on our device with nPM1300.
 Thank you all.

**Comment by jasr93** [comment]
Compliance check has been fixed.
Added code suggestion too.

**Review by ubieda — APPROVED** [review]

**Review by nordic-auko — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Pull request merged**